lib/config/Configuration.js

```diff
@@ -0,0 +1,332 @@
+	 * Directory to store framework artifacts and metadata in.
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getFrameworkDir() {
+		return this.#frameworkDir;
+	}
+
+	/**
+	 * Directory to store artifacts in.
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getArtifactsDir() {
+		return this.#artifactsDir;
+	}
+
+	/**
+	 * Directory to packages in
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getPackagesDir() {
+		return this.#packagesDir;
+	}
+
+	/**
+	 * Directory to store metadata in
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getMetadataDir() {
+		return this.#metadataDir;
+	}
+
+	/**
+	 * Directory to store staging artifacts in
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getStagingDir() {
+		return this.#stagingDir;
+	}
+
+	/**
+	 * Lockfiles directory
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getLockDir() {
+		return this.#lockDir;
+	}
+
+	/**
+	 * Cache directory
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getCacheDir() {
+		return this.#cacheDir;
+	}
+
+	/**
+	 * @public
+	 * @returns {object} The configuration in a JSON format
+	 */
+	toJSON() {
+		return {
+			cwd: this.#cwd,
+			sources: this.#sources,
+			version: this.#version,
+			cacheMode: this.#cacheMode,
+			ui5HomeDir: this.#ui5HomeDir,
+			snapshotEndpointUrl: this.#snapshotEndpointUrl,
+			frameworkDir: this.#frameworkDir,
+			artifactsDir: this.#artifactsDir,
+			packagesDir: this.#packagesDir,
+			metadataDir: this.#metadataDir,
+			stagingDir: this.#stagingDir,
+			lockDir: this.#lockDir,
+			cacheDir: this.#cacheDir,
+		};
+	}
+}
+
+export default Configuration;
+
+/**
+ * Create Configuration from JSON file
+ *
+ * @public
+ * @static
+ * @param {string} [filePath="~/.ui5rc"] Path to configuration JSON file
+ * @returns {@ui5/project/config/Configuration} Configuration instance
+ */
+export async function fromFile(filePath) {
+	filePath = filePath || path.join(os.homedir(), ".ui5rc");
+
+	const {default: fs} = await import("graceful-fs");
+	const {promisify} = await import("node:util");
+	const readFile = promisify(fs.readFile);
+	let config;
+	try {
+		const fileContent = await readFile(filePath);
+		config = JSON.parse(fileContent);
+	} catch (err) {
+		if (err.code === "ENOENT") { // "File or directory does not exist"
+			config = {};
+		} else {
+			throw err;
+		}
+	}
+	return new Configuration(config);
+}
+
+/**
+ * Save Configuration to a JSON file
+ *
+ * @public
+ * @static
+ * @param {string} [filePath="~/.ui5rc"] Path to configuration JSON file
+ * @param {@ui5/project/config/Configuration} config Configuration to save
+ */
+export async function saveConfig(filePath, config) {
+	filePath = filePath || path.join(os.homedir(), ".ui5rc");
+
+	const {default: fs} = await import("graceful-fs");
+	const {promisify} = await import("node:util");
+	const writeFile = promisify(fs.writeFile);
+
+	return writeFile(filePath, JSON.stringify(config.toJSON()));
+}
+
+
+export async function resolveSnapshotEndpointUrl(settingsXML, skipConfirmation) {
```

Comment:
We've discussed this and need to look at it from another perspective- the end consumer i.e. the developers.
We do not expect them to have Maven installed. 
With the potential CLI implementation of the configuration settings, addressed here: https://github.com/SAP/ui5-tooling/blob/rfc-configuration/rfcs/0013-configuration.md#changing-configuration, it might become even redundant. Developers, when setting their environment, would eventually need just to execute once a single CLI command like `ui5 config set snapshotUrl https://.....` 

If we decide that we still need this revolver, its place must be somewhere in the mvn/Installer where it actually be in use.